tests/vegalite/v5/test_params.py

```diff
@@ -167,6 +167,24 @@ def test_selection_condition():
     assert dct["encoding"]["size"]["value"] == 10
 
 
+def test_selection_interval_value_typing() -> None:
+    import datetime as dt
+
+    window_date = dt.date(2005, 1, 1), dt.date(2009, 1, 1)
+    window_other = (0, 999)
+
+    a = alt.selection_interval(encodings=["x"], value={"x": window_date}).to_dict()
+    b = alt.selection_interval(encodings=["y"], value={"y": window_other}).to_dict()
+    a_b = alt.selection_interval(  # type: ignore[type-var]
+        encodings=["x", "y"],
+        value={"x": window_date, "y": window_other},  # pyright: ignore[reportArgumentType]
+    ).to_dict()
```

Comment:
### TODO
- Turn these into unused ignore

Commit message (https://github.com/vega/altair/pull/3653/commits/8c664208c7817da90621097119ffb80b5471f007):

> Need to fix this, but adding this first to demonstrate the issue.
> Currently the way I've typed it prevents using different types for each encoding.
> 
> The goal is for each encoding to restrict types independently.
> E.g. here `x` is **only** `date`, `y` is **only** `float`

Reply:
Fixed in (https://github.com/vega/altair/pull/3653/commits/901e9b38d7506925a3b8bb7cd16972471db77918)